### PR TITLE
[ADD] travis: Add script to set Travis variables when running on GitLab

### LINKED ISSUE #266: gitlab is not running the special lint when is a pull request
Travis set the following [environment variables](https://docs.travis-ci.com/user/environment-variables/)
 - `TRAVIS_PULL_REQUEST - The pull request number if the current job is a pull request, “false” if it’s not a pull request.` 
 - `TRAVIS_BRANCH - this is the name of the branch targeted by the pull request. E.g. 11.0` 

Gitlab works a little different and then these checks are not emitted.

Differences between gitlab and github:
 - If there is a PR from vauxoo-dev:11.0-devx-moy to vauxoo:11.0 the repository cloned is: `git clone vauxoo` and in order to get the PR commits uses the `refs` defined by github: `git fetch -p origin +refs/pull/*/head:refs/pull/origin/*` after that creates a checkout to `pull/origin/${TRAVIS_PULL_REQUEST}`

 - gitlab creates the pipeline before to create a MR then I think that the remote of the target repository is not defined yet and is not cloned the target repo. (I think so...) 

Then maybe we will need to define a environment variable (or use one defined from gitlab) in order to add the remote to target branch. Same case for the target branch.

### PR DESCRIPTION
**Pull request by luisg123v**
This adds a new script to set Travis CI's environment variables when a
project is being built on GitLab CI.

Closes #266

Dummy PR/MRs:
- [GitLab](https://git.vauxoo.com/vauxoo/enterprise-mx/merge_requests/310): Show how this script would be used, along with how variables are set after it runs.
- [GitHub](https://github.com/Vauxoo/mexico/pull/1171): Print how Travis CI variables are set on a regular case, along with other interesting git information.

### CODE REVIEW

**Comment by luisg123v** [comment]
@moylop260 
Do you like this approach?

I Haven't extensively tested it yet, I'd like your approval before furter working on this.

**Comment by moylop260** [comment]
I like it
Where have you planned invoke this script?

**Comment by luisg123v** [comment]
My idea is to source it in projects running  under GitLab CI, from the file `.gitlab-ci.yml`

**Comment by moylop260** [comment]
@ruiztulio 
What do you think?

File: travis/travis_gitlab_translate_env [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    export TRAVIS_PULL_REQUEST="unknown"
+    export TRAVIS_PULL_REQUEST_BRANCH="${CI_COMMIT_REF_NAME}"
+    export TRAVIS_PULL_REQUEST_SHA="${CI_COMMIT_SHA}"
+    export TRAVIS_PULL_REQUEST_SLUG="${CI_PROJECT_PATH}"
+fi
+export TRAVIS_REPO_SLUG=${CI_PROJECT_PATH}
+if [[ $(whoami) == "root" ]]; then
+    exportTRAVIS_SUDO=true"
+else
+    exportTRAVIS_SUDO=false"
```
Comment (ruiztulio) [inline travis/travis_gitlab_translate_env]: @luisg123v I think you missed a space here and in line 65:

```export TRAVIS_SUDO="false"```

**Comment by ruiztulio** [comment]
@moylop260 @luisg123v Other than the typos I commented LGTM, in this way we do not have to map the env vars from travis -> gitlab manually to use the MQT tools, I think we just need to test it properly

File: travis/travis_gitlab_translate_env (hunk 1 above)
Comment (luisg123v) [reply travis/travis_gitlab_translate_env]: Done, and quoted all variable values.

Comment (ruiztulio) [reply travis/travis_gitlab_translate_env]: @luisg123v I don't know if it is only me, but I still don't see the missing space:

a

Comment (luisg123v) [reply travis/travis_gitlab_translate_env]: :laughing: Sorry, done.

**Comment by luisg123v** [comment]
@moylop260 
I think this is ready for review.

I updated the PR description to include dummys for both GitLab and GitHub, to show how this would be used and to print debug information.

What do you think?

**Comment by moylop260** [comment]
Could you add the list of prs here?

Note: Use vx repos

**Comment by luisg123v** [comment]
Yes, sure.:
- [GitLab](https://git.vauxoo.com/vauxoo/enterprise-mx/merge_requests/310): Show how this script would be used, along with how variables are set after it runs.
- [GitHub](https://github.com/Vauxoo/mexico/pull/1171): Print how Travis CI variables are set on a regular case, along with other interesting git information.

Tey are VX repos (enterprise-mx and mexico)

**Comment by ruiztulio** [comment]
@luisg123v But the one from gitlab is red, to make a proper test use a repo that already have a green pipeline.

Also DON'T print secret/private variables please, as you may imagine it is not a good practice:

a

**Comment by luisg123v** [comment]
@moylop260 
Got it, done at: https://git.vauxoo.com/vauxoo/project-template/merge_requests/10

**Pull request merged**